core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -619,11 +620,16 @@ protected Type visitDereferenceExpression(DereferenceExpression node, StackableA
             RowType rowType = (RowType) baseType;
             String fieldName = node.getField().getValue();
 
+            int index = -1;
             Type rowFieldType = null;
             for (RowType.Field rowField : rowType.getFields()) {
-                if (fieldName.equalsIgnoreCase(rowField.getName().orElse(null))) {
+                String rowFieldName = rowField.getName().orElse(null);
```

Comment:
Inline `rowFieldName`.

Reply:
Copy that. I've addressed your latest comments. Thanks !